core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/imports/ImportKeystoresHandler.java

```diff
@@ -0,0 +1,248 @@
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final int SUCCESS = 200;
+  public static final int BAD_REQUEST = 400;
+  public static final int SERVER_ERROR = 500;
+  private static final ObjectMapper YAML_OBJECT_MAPPER = new ObjectMapper(new YAMLFactory());
+
+  private final ObjectMapper objectMapper;
+  private final Path keystorePath;
+  private final Optional<SlashingProtection> slashingProtection;
+  private final ArtifactSignerProvider artifactSignerProvider;
+
+  public ImportKeystoresHandler(
+      final ObjectMapper objectMapper,
+      final Path keystorePath,
+      final Optional<SlashingProtection> slashingProtection,
+      final ArtifactSignerProvider artifactSignerProvider) {
+    this.objectMapper = objectMapper;
+    this.keystorePath = keystorePath;
+    this.slashingProtection = slashingProtection;
+    this.artifactSignerProvider = artifactSignerProvider;
+  }
+
+  @Override
+  public void handle(RoutingContext context) {
+    // API spec - https://github.com/ethereum/keymanager-APIs/tree/master/flows#import
+    final RequestParameters params = context.get("parsedParameters");
+    final ImportKeystoresRequestBody parsedBody;
+    try {
+      parsedBody = parseRequestBody(params);
+    } catch (final IllegalArgumentException | JsonProcessingException e) {
+      handleInvalidRequest(context, e);
+      return;
+    }
+
+    // check that keystores have matching passwords
+    if (parsedBody.getKeystores().size() != parsedBody.getPasswords().size()) {
+      context.fail(BAD_REQUEST);
+      return;
+    }
+
+    // no keystores passed in, nothing to do, return 200 and empty response.
+    if (parsedBody.getKeystores().isEmpty()) {
+      try {
+        context
+            .response()
+            .putHeader(CONTENT_TYPE, JSON_UTF_8)
+            .setStatusCode(SUCCESS)
+            .end(
+                objectMapper.writeValueAsString(
+                    new ImportKeystoresResponse(Collections.emptyList())));
+      } catch (JsonProcessingException e) {
+        context.fail(SERVER_ERROR, e);
+      }
+      return;
+    }
+
+    // extract pubkeys to import first
+    final List<String> pubkeysToImport = new ArrayList<>();
+    try {
+      pubkeysToImport.addAll(
+          parsedBody.getKeystores().stream()
+              .map(json -> new JsonObject(json).getString("pubkey"))
+              .map(
+                  key ->
+                      key.startsWith("0x")
+                          ? key
+                          : "0x" + key) // always use 0x prefix for comparisons
+              .collect(Collectors.toList()));
+    } catch (Exception e) {
+      context.fail(BAD_REQUEST, e);
+      return;
+    }
+
+    // read slashing protection data if present and import data matching keys to import only
+    if (slashingProtection.isPresent()
+        && !StringUtils.isEmpty(parsedBody.getSlashingProtection())) {
```

Comment:
done

Reply:
I see don't see the https://github.com/FasterXML/jackson-modules-java8 module now either. Thought it was there. Minor quibble though so I think this can stay as it is